src/core/algorithms/md/hymd/lattice/md_lattice.cpp

```diff
@@ -471,17 +473,21 @@ std::size_t MdLattice::MdRefiner::Refine() {
             ++removed;
             continue;
         }
-        DESBORDANTE_ASSUME(node_info_.rhs->begin[rhs_index] == kLowestCCValueId &&
+        DESBORDANTE_ASSUME(rhs.begin[rhs_index] == kLowestCCValueId &&
                            new_ccv_id != kLowestCCValueId);
-        node_info_.rhs->Set(rhs_index, new_ccv_id);
+        rhs.Set(rhs_index, new_ccv_id);
     }
     lattice_->Specialize(GetLhs(), *pair_comparison_result_, invalidated_.GetInvalidated());
+    if (rhs.IsEmpty() && node_info_.node->IsEmpty()) {
+        lattice_->TryDeleteEmptyNode(GetLhs());
```

Comment:
We know for certain that the node exists at this point, check on line 580 is extraneous.